test/e2e_node/density_test.go

```diff
@@ -0,0 +1,490 @@
+
+				// verify resource
+				By("Verifying resource")
+				verifyResource(f, testArg, rm)
+
+				// delete pods
+				By("Deleting a batch of pods")
+				deleteBatchPod(f, pods)
+
+				// tear down cadvisor
+				Expect(f.Client.Pods(ns).Delete(cadvisorPodName, api.NewDeleteOptions(30))).
+					NotTo(HaveOccurred())
+
+				Eventually(func() error {
+					return checkPodDeleted(f, cadvisorPodName)
+				}, 10*time.Minute, time.Second*3).Should(BeNil())
+			})
+		}
+	})
+
+	Context("create a sequence of pods", func() {
+		densityTests := []DensityTest{
+			{
+				podsNr:   10,
+				bgPodsNr: 10,
+				cpuLimits: framework.ContainersCPUSummary{
+					stats.SystemContainerKubelet: {0.50: 0.10, 0.95: 0.12},
+					stats.SystemContainerRuntime: {0.50: 0.16, 0.95: 0.20},
+				},
+				memLimits: framework.ResourceUsagePerContainer{
+					stats.SystemContainerKubelet: &framework.ContainerResourceUsage{MemoryRSSInBytes: 40 * 1024 * 1024},
+					stats.SystemContainerRuntime: &framework.ContainerResourceUsage{MemoryRSSInBytes: 300 * 1024 * 1024},
+				},
+				podStartupLimits: framework.LatencyMetric{
+					Perc50: 1500 * time.Millisecond,
+					Perc90: 2500 * time.Millisecond,
+					Perc99: 3500 * time.Millisecond,
+				},
+			},
+			{
+				podsNr:   10,
+				bgPodsNr: 30,
+				cpuLimits: framework.ContainersCPUSummary{
+					stats.SystemContainerKubelet: {0.50: 0.12, 0.95: 0.15},
+					stats.SystemContainerRuntime: {0.50: 0.22, 0.95: 0.27},
+				},
+				memLimits: framework.ResourceUsagePerContainer{
+					stats.SystemContainerKubelet: &framework.ContainerResourceUsage{MemoryRSSInBytes: 40 * 1024 * 1024},
+					stats.SystemContainerRuntime: &framework.ContainerResourceUsage{MemoryRSSInBytes: 300 * 1024 * 1024},
+				},
+				podStartupLimits: framework.LatencyMetric{
+					Perc50: 1500 * time.Millisecond,
+					Perc90: 1500 * time.Millisecond,
+					Perc99: 1600 * time.Millisecond,
+				},
+			},
+		}
+
+		for _, testArg := range densityTests {
+			itArg := testArg
+			It(fmt.Sprintf("latency/resource should be within limit when create %d pods with %d background pods",
+				itArg.podsNr, itArg.bgPodsNr), func() {
+				bgPods := newTestPods(itArg.bgPodsNr, ImageRegistry[pauseImage], "background_pod")
+				testPods := newTestPods(itArg.podsNr, ImageRegistry[pauseImage], podType)
+
+				createCadvisorPod(f)
+				rm := NewResourceCollector(monitoringInterval)
+
+				By("Creating a batch of background pods")
+				// creatBatch is synchronized
+				// all pods are running when it returns
+				f.PodClient().CreateBatch(bgPods)
+
+				//time.Sleep(sleepBeforeCreatePods)
+
+				// starting resource monitoring
+				rm.Start()
+
+				// do a sequential creation of pod (back to back)
+				batchlag, e2eLags := createBatchPodSequential(f, testPods)
+
+				rm.Stop()
+
+				// verify latency
+				By("Verifying latency")
+				verifyLatency(batchlag, e2eLags, itArg)
+
+				// verify resource
+				By("Verifying resource")
+				verifyResource(f, testArg, rm)
+
+				// delete pods
```

Comment:
same question I had in a different file: why does the test need to clean up manually (as opposed to waiting for the namespace controller to nuke all the resources.